Add ruby-1-b (#58)
exercism/research_experiment_1: exercises/ruby-1-b/README.md

## Introduction

This is Part 1 of our research into how people write Ruby differently.
This exercise should take 10-15 minutes. There is no right or wrong way
to approach it - just do what feels the most natural to you. Feel free
to use books, Google or Stack Overflow, just like you would if you were
programming normally, but please don't actively look for a solution to
the problem as this will negatively affect the research.

## Instructions

You are given a collection of library books, and your job is to import them into
the library's catalog system. The books are in a text file which provides a title,

> ISBN, author, and publishing date for each book.

Author: ```suggestion
ISBN, author, and publishing year for each book.
```

> Write the methods that will import the data for the books from the text file to

Reviewer: I'm missing the bit that explains what code the student is expected to write.

the library's catalog system.

The data is stored in plain text in a file called `books.txt`. One record for each line,
and the `;` character is used as a delimiter. The first two lines look like this:

> 

Reviewer: Could you perhaps also add some example code?

> ## Instructions

Reviewer 2: This README needs to be a full spec of what needs to happen. People cannot see the tests when solving this exercise. So things like `.published` and the other methods with their meanings and things like `ArgumentError` all need explicitly specifying in here.

You are given a collection of library books, and your job is to import them into
the library's catalog system. The books are in the constant `Library::BOOKS`, and
each book has the following columns:
title, ISBN, author, and publishing year.

You need to write the logic for finding a book, adding new copies to the library's

> inventory, lending a book, and returning a book

Reviewer: ```suggestion
inventory, lending a book, and returning a book.
```

> A book should be findable with it's ISBN

Reviewer: ```suggestion
A book should be findable with it's ISBN:
```

```
book = Library.new.find("9780191604928")
```

> And it should have getters for the attribute values provided in `Library::BOOKS`

Reviewer: ```suggestion
And it should have getters for the book's attributes:
```

```
book.author

> => "Mark Twain"

Reviewer: This is slightly more consistent with the [ruby-1-a README](https://github.com/exercism/research_experiment_1/tree/master/exercises/ruby-1-a):

```suggestion
// Returns: "Mark Twain"
```

book.title

> => "The Adventures of Tom Sawyer"

Reviewer: This is slightly more consistent with the [ruby-1-a README](https://github.com/exercism/research_experiment_1/tree/master/exercises/ruby-1-a):

```suggestion
// Returns: "The Adventures of Tom Sawyer"
```
